prowler/providers/aws/lib/security_hub/security_hub.py

```diff
@@ -284,3 +310,114 @@ def _send_findings_in_batches(
+                session=session,
+                aws_account_id=aws_account_id,
+                aws_partition=aws_partition,
+            ).keys()
+            disabled_regions = all_regions - enabled_regions
+            if regions:
+                if not any(region in enabled_regions for region in regions):
+                    logger.warning(
+                        f"Prowler integration is not enabled in regions: {regions - enabled_regions}."
+                    )
+                    invalid_region_error = SecurityHubInvalidRegionError(
+                        message="Given regions have not Security Hub enabled."
+                    )
+                    if raise_on_exception:
+                        raise invalid_region_error
+                    return SecurityHubConnection(
+                        is_connected=False,
+                        error=invalid_region_error,
+                        enabled_regions=enabled_regions,
+                        disabled_regions=disabled_regions,
+                    )
+                else:
+                    logger.info(
+                        f"Prowler integration is enabled in regions: {list(regions)}."
+                    )
+                    return SecurityHubConnection(
+                        is_connected=True,
+                        error=None,
+                        enabled_regions=enabled_regions,
+                        disabled_regions=disabled_regions,
+                    )
+
+            if len(enabled_regions) == 0:
+                logger.warning(
+                    "No regions were found to with the Security Hub integration enabled."
+                )
+                no_enabled_regions_error = SecurityHubNoEnabledRegionsError(
+                    message="No regions were found to with the Security Hub integration enabled."
+                )
+                if raise_on_exception:
+                    raise no_enabled_regions_error
+                return SecurityHubConnection(
+                    is_connected=False,
+                    error=no_enabled_regions_error,
+                    enabled_regions=enabled_regions,
+                    disabled_regions=disabled_regions,
+                )
+            else:
+                logger.info(
+                    f"Security Hub is enabled in the following regions: {', '.join(enabled_regions)}."
+                )
+                return SecurityHubConnection(
+                    is_connected=True,
+                    error=None,
+                    enabled_regions=enabled_regions,
+                    disabled_regions=disabled_regions,
+                )
+
+        except Exception as error:
+            logger.error(
+                f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+            exception_error = SecurityHubTestConnectionError(
+                message="Failed to test connection on the Security Hub integration"
```

Comment:
We need to add the original exception here, right?

Reply:
Good catch, I've added it in the last commit